R/compat-vctrs.R

```diff
@@ -0,0 +1,102 @@
+# based on https://github.com/tidymodels/workflowsets/blob/main/R/compat-vctrs.R
+
+# ------------------------------------------------------------------------------
+
+# `vec_restore()`
+#
+# Called at the end of `vec_slice()` and `vec_ptype()` after all slicing has
+# been done on the proxy object.
+
+#' @export
+vec_restore.dribble <- function(x, to, ...) {
+   dribble_maybe_reconstruct(x)
+}
+
+# ------------------------------------------------------------------------------
+
+# `vec_ptype2()`
+#
+# When combining two dribbles together, `x` and `y` will be zero-row slices
+# which should always result in a new dribble, as long as
+# `df_ptype2()` can compute a common type.
+#
+# Combining a dribble with a tibble/data.frame will only ever happen if
+# the user calls `vec_c()` or `vec_rbind()` with one of each of those inputs.
+# Although I could probably make this work, it feels pretty weird and exotic
+# and I think that user, if they even exist, should just turn that "other"
+# tibble/data.frame into a dribble first.
+# So I'll follow workflowsets and not attempt to return dribble for these
+# combinations.
```

Comment:
I find that keeping it as simple as possible (as you've done here) is much more rewarding. Trying to be too "smart" always ended up being extremely buggy and fragile.